quantum/wear_leveling/wear_leveling.c

```diff
@@ -582,26 +639,26 @@ wear_leveling_status_t wear_leveling_write(const uint32_t address, const void *v
     memcpy(&wear_leveling.cache[address], value, length);
 
     // Unlock the backing store
-    bool ret = backing_store_unlock();
+    bool ret = wear_leveling_unlock();
     if (!ret) {
         // If the unlock failed, re-lock and exit with failure
-        backing_store_lock();
+        wear_leveling_lock();
         return WEAR_LEVELING_FAILED;
     }
 
     // Perform the actual write
     wear_leveling_status_t status = wear_leveling_write_raw(address, value, length);
     if (status != WEAR_LEVELING_SUCCESS) {
         // If the write failed, re-lock and exit with failure
-        backing_store_lock();
+        wear_leveling_lock();
         return status;
     }
 
     // Consolidate the cache + write log if required
     status = wear_leveling_consolidate_if_needed();
 
     // Lock the backing store
-    ret = backing_store_lock();
+    ret = wear_leveling_lock();
     return ret ? status : WEAR_LEVELING_FAILED;
```

Comment:
if `wear_leveling_consolidate_if_needed()` happens to consolidate then the flash will be locked by that function. Thus `wear_leveling_lock()` will return `false` this should result in `WEAR_LEVELING_FAILED` although there was no error? 

If so, I guess that there are multiple ways to handle the situation, maybe the `lock` should not be called when `wear_leveling_consolidate_if_needed` returns `WEAR_LEVELING_CONSOLIDATED`. Another option would be to check if `wear_leveling_lock` returns "already locked" wich would not be en error in this case. Or something completely different...